File: LeetCode/Easy/3136-Valid_Word/3136-ValidWord.js
/**
 * @param {string} word
 * @return {boolean}
 */
var isValid = function (word) {
    if (word.length < 3)
        return false;

    let vowel = false;
    let consonant = false;

    for (const c of word) {
        if (/[a-zA-Z]/.test(c)) {
            let low = c.toLowerCase();
            if (low === 'a' || low === 'e' || low === 'i' || low === 'o' || low === 'u')
                vowel = true;
            else
                consonant = true;
        } else if (!/\d/.test(c))
            return false;
    }

    return vowel && consonant;
}

File: LeetCode/Easy/3136-Valid_Word/3136-ValidWord.cpp
#include <string>

class Solution
{
public:
    bool isValid(string word)
    {
        if (word.size() < 3)
            return false;

        bool vowel = false;
        bool consonant = false;

        for (auto c : word)
        {
            if (!isalnum(c))
                return false;

            if (isalpha(c))
            {
                char low = tolower(c);
                if (low == 'a' || low == 'e' || low == 'i' || low == 'o' || low == 'u')
                    vowel = true;
                else
                    consonant = true;
            }
        }

        return vowel && consonant;
    }
};